### PR TITLE
Update bin/fmt to Handle Git Worktrees

### PR DESCRIPTION
**Pull request by gwendolyngoetz**
### Description

The formatter code passes in the path to the code to docker. This would work fine if the code is cloned in the more common way with `git clone` and then selecting the active branch.

However if the repository were cloned as a bare repository and branches added as a separate `git worktree` the formatter wouldn't have access to the git repository metadata and would thus fail. This change checks if on a worktree or not and passes in the path to the git metadata if needed.

#### General

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >

### CODE REVIEW

**Review by bion — CHANGES_REQUESTED** [review]
formatter workflow is failing

File: bin/fmt [hunk 1]
```diff
@@ -28,7 +28,45 @@ if [[ -n "${CI}" ]]; then
   interactive_flag=""
 fi
 
-docker run --rm ${interactive_flag} \
-  -v "$(pwd):/code" \
-  civiform/formatter \
-  "$@"
+gitdir="$(git rev-parse --show-toplevel)"
+repotype="not-worktree"
```
Comment (bion) [inline bin/fmt:32]: **praise:** please see styleguide for variable naming convention https://google.github.io/styleguide/shellguide.html

```suggestion
readonly GIT_DIR="$(git rev-parse --show-toplevel)"
repo_type="not-worktree"
```

**Review by bion — CHANGES_REQUESTED** [review]

File: bin/fmt [hunk 2]
```diff
@@ -28,7 +28,45 @@ if [[ -n "${CI}" ]]; then
...
+# is a branch checked out in its own folder. For this we pass in the path of the worktree parent so
+# the fmt command will has access to other branches when it does a diff.
+# Example:
+#   > git clone --bare git@github.com:civiform/civiform.git
+#   > cd ./civiform
+#   > git worktree add -B new-branch-name ./new-branch-name
+#   > cd ./new-branch-name
+#   > bin/fmt
+elif [[ "${repotype}" == "worktree" ]]; then
+  worktree_root="$(dirname "${gitdir}")"
```
[the remark this answers is not in the document]
Comment (gwendolyngoetz) [reply bin/fmt]: Skipping this since local is only valid within functions.

File: bin/fmt (hunk 1 above)
Comment (gwendolyngoetz) [reply bin/fmt]: Done

File: bin/fmt (hunk 2 above)
[the remark this answers is not in the document]
Comment (bion) [reply bin/fmt]: 🤦🏻 woops, thank you

**Review by bion — APPROVED** [review]

**Pull request merged**